server/bin/pbench-index.py

```diff
@@ -435,6 +461,8 @@ def sigquit_handler(*args):
                         "Indexing interrupted by SIGTERM, terminating"
                     )
                     break
+                except DatasetTransitionError:
+                    idxctx.logger.exception("Dataset state error")
```

Comment:
Perhaps move this to a `try/except` immediately around the `dataset.advance(Status.INDEXED)` call?

Reply:
I considered it. I think maybe I was lazy. I'll look at it again.